src/main/webapp/dev/js/main/instructorCourseEnrollPage.js

```diff
@@ -160,6 +160,27 @@ function getAjaxStudentList(ajaxPreloadImage) {
     });
 }
 
+/**
+ * Gets enroll status through an AJAX request.
+ * @returns {Promise} the state of the result from the AJAX request
+ */
+function getAjaxEnrollStatus() {
+    return new Promise((resolve, reject) => {
+        const $spreadsheetForm = $('#student-spreadsheet-form');
+        $.ajax({
+            type: 'POST',
+            url: '/page/instructorCourseEnrollAjaxSave',
+            cache: false,
+            data: {
+                courseid: $spreadsheetForm.children(`input[name='${ParamsNames.COURSE_ID}']`).val(),
+                enrollstudents: $spreadsheetForm.find('#enrollstudents').val(),
```

Comment:
These POST parameters at least lacks the CSRF token to prevent CSRF attack. What are the parameters that are sent in the original enrollment form?

Reply:
The original enrollment form does include a `SESSION_TOKEN` variable. Does this mean that `getAjaxStudentList` should also pass the session token in the data object as well?

Reply:
Yes, this new method should include all fields sent by the original enrollment form, unless there are reasons not to include them.

Reply:
I've checked that the backend does recognize the same token value which is present in the JSP file, which is why the already implemented `InstructorCourseEnrollAjaxPageAction` is able to initialize `InstructorCourseEnrollAjaxPageData` with `sessionToken` as one of the constructor's parameters. Guess we don't need to explicitly pass this token as a POST parameter.

Reply:
@tran-tien-dat I've checked that the backend does recognize the same token value which is present in the JSP file, which is why the already implemented `InstructorCourseEnrollAjaxPageAction` is able to initialize `InstructorCourseEnrollAjaxPageData` with `sessionToken` as one of its constructor's parameters. Guess we don't need to explicitly pass this token as a POST parameter.

Reply:
I don't get what you mean. Are you saying that everything still works without having to include the session token as a POST parameter?

Reply:
Yup that's right.

Reply:
Including the `sessionToken` in the POST request is not for the application to work, but for security reason. In particular it is to prevent a CSRF attack. You can read about it [here](https://github.com/se-edu/learningresources/blob/master/contents/security/crossSiteRequestForgery/crossSiteRequestForgery.md)

This is why I made [this comment](https://github.com/TEAMMATES/teammates/pull/9017/files#r207743872). After you have understood the article I linked above, you can trace the usage of `PAGES_REQUIRING_ORIGIN_VALIDATION` which you modified in the comment I linked. That would show you how teammates implement the mitigation I suggested in the article.

So in conclusion, you need to put this new AJAX endpoint back to `PAGES_REQUIRING_ORIGIN_VALIDATION` and POST the `sessionToken`